### ISSUE TITLE
Enhancement on date format handling

### ISSUE THREAD

**Issue by rpiotaix** [highcharts/highcharts#5418]
My issue is almost like #3435, but i'm creating a new one as i discuss a wider range of topics.

I stumbled across a limitation when trying to internatonalize Highcharts (i've got a English/French/maybe Italian website). I found that it was possible to define the month/days names using setOptions({lang:...}). For the default date format ('%Y-%m-%d %H:%M:%S'), it is hardcoded into the 'dateFormat' function.
In addition, the default format for the tooltip gives me dates like 'mardi, juin 21, 11:10' which is the american way of writing dates and should be 'mardi 21 juin, 11:10' for the french locale.

Anyway, i noticed that:
- The format function (for date and integer formats in labels/tooltips/etc.) follow (i quote) "a subset of the rules of Python's String.format method". Fine.
- The format function use 'format single' That is "Similar to sprintf, without the % prefix". Uh.
- The date formats are inspired by PHP strftime function ?!?!

In addition, the Highcharts.dateFormat is just an 'export' of the function defined in a narrower scope inside Highcharts and is not used internally. So redefining it will not change anything internally.

So, basically, i was looking for a way of changing the formats used to be able to use everywhere in my application the same formats as moment.js that i'm using for handling dates. (Hell, why not even use momentjs for dates in highcharts as well. But this is an other topic.)

I can help implement these new functionalities if needed.

**Comment by TorsteinHonsi** [comment]
I don't think we really need an enhancements, you just have to define the language specific date formats in the appropriate places. Like for the tooltip, it would be in [tooltip.dateTimeLabelFormats](http://api.highcharts.com/highcharts#tooltip.date). See http://jsfiddle.net/highcharts/nfzhvduv/1/.

I admit that we should consider moving this to the language object, since the languages not only have different words for weeks or months and puncutation, but also different orders of the full date.

**Comment by rpiotaix** [comment]
Ok for the localization. Will do it this way.

What about doing as #3284 for dateFormat?

**Comment by TorsteinHonsi** [comment]
> What about doing as #3284 for dateFormat?

That will be possible in Highcharts 5. But I don't really see when you would need to override it? If it is just to add new formats, use [dateFormats](http://api.highcharts.com/highcharts#Highcharts.dateFormats).

**Comment by TorsteinHonsi** [comment]
> The format function use 'format single' That is "Similar to sprintf, without the % prefix". Uh.

Note that this is internal only. Not meant for the outside :)

**Comment by rpiotaix** [comment]
> But I don't really see when you would need to override it?
I was looking for an easy way of integrating highcharts with momentjs. One part can be done using Highcharts.Date but the formatting was an other problem.

I think we can close this issue as I will use the translation component to get the date formats according to the locale that is being used: this is a good enough solution for now.

**Comment by TorsteinHonsi** [comment]
Okay, thanks!
